File: Majo/Act-Instructor_Henry/lenguajes/dart/junio/07-06-24.dart
import 'dart:io';

void main(){
  
  // Variables para >=5000
  int cont = 0;
  int acum = 0;

  // Variables para <5000
  int cont2 = 0;
  int acum2 = 0;

  print('Ingresa el número de personas participando');
  int num = int.parse(stdin.readLineSync()!);
  
  for (var i = 1; i <= num; i++) {
    
    print('Ingresa la cantidad que deseas aportar');
    int Var = int.parse(stdin.readLineSync()!); 
  

    if (Var >= 5000 ) {

     cont = cont + 1;
     acum = acum + Var;

    } else {

     cont2 = cont2 + 1;
     acum2 = acum2 + Var;

    }
  
  }

    int total = acum + acum2;
    double prom = total / num;
    double prom2 = acum /cont;
    double prom3 = acum2 / cont2;


  print('---------------------------------------------------------------------------');
  print(' ');
  print('-La cantidad total recogida es: $total.');
  print(' ');
  print('-El promedio del total recogido es: $prom.');
  print(' ');
  print('---------------------------------------------------------------------------');
  print('-Personas que aportaron 5.000  o más: $cont.');
  print('-Cantidad total de los que aportaron 5.000 o más: $acum.');
  print('-Promedio de aporte de los que dieron 5.000 o más $prom2.');
  print(' ');
  print('---------------------------------------------------------------------------');
  print('-Personas que aportaron menos de 5.000: $cont2.');
  print('-Cantidad de los que aportaron menos de 5.000: $acum2.');
  print('-Promedio de aporte de los que dieron menos de 5.000: $prom3.');
  print(' ');
  
}

File: Majo/Act-Instructor_Henry/lenguajes/ruby/junio/07-06-24.rb
# Variables para >=5000
cont = 0
acum = 0

# Variables para <5000
cont2 = 0
acum2 = 0

puts "Ingresa el número de personas participando"
num = gets.chomp.to_i

for i in 1..num
    puts "Ingresa la cantidad que deseas aportar"
    var = gets.chomp.to_f
    
    if var >= 5000
        cont = cont + 1
        acum = acum + var
    else
        cont2 = cont2 + 1
        acum2 = acum2 + var
    end
end

total = acum +acum2
prom = total / num
prom2 = acum / cont
prom3 = acum2 / cont2

puts "---------------------------------------------------------------------------"
puts " "
puts "-La cantidad total recogida es: #{total}."
puts " "
puts "-El promedio del total recogido es: #{prom}"
puts " "
puts "---------------------------------------------------------------------------"
puts "-Personas que aportaron 5.000 o más: #{cont}."
puts "-Cantidad total de los que aportaron 5.000 o más: #{acum}."
puts "-Promedio de aporte de los que dieron 5.000 o más: #{prom2}."
puts " "
puts "---------------------------------------------------------------------------" 
puts "-Personas que aportaron menos de 5.000: #{cont2}."
puts "-Cantidad total de los que aportaron menos de 5.000: #{acum2}."
puts "-Promedio de aporte de los que dieron menos de 5.000: #{prom3}."
puts " "